admin/ops/setup_dev_bucket.py

```diff
@@ -0,0 +1,342 @@
+        raise ValueError("Only one of `include` and `exclude` may be provided")
+    datasets = get_subfolders(bucket=PROD_RECIPES_BUCKET, prefix="datasets/")
+    if include:
+        datasets = include
+    if exclude:
+        datasets = [ds for ds in datasets if ds not in exclude]
+
+    for ds in datasets:
+        clone_recipe(target_bucket, ds)
+
+
+def clone_data_products(
+    target_bucket: str,
+    data_products: list[str],
+    include_recipe_datasets: bool = True,
+):
+    """
+    Clone latest published version of each provided data product
+    If `include_recipe_datasets`, input datasets for all provided products are collected and then cloned (latest versions only)
+
+    This is a good target if setting up a dev bucket to work on enhancements for a data product
+    """
+    if include_recipe_datasets:
+        recipe_datasets = []
+        for product in data_products:
+            recipe = plan.recipe_from_yaml(
+                ROOT_PATH / "products" / product.strip("db-") / "recipe.yml"
+            )
+            recipe_datasets += [ds.name for ds in recipe.inputs.datasets]
+        for ds in set(recipe_datasets):
+            clone_recipe(target_bucket, ds)
+
+    for product in data_products:
+        key = PublishKey(product=product, version="latest")
+        copy_folder_to_dev_bucket(
+            source_bucket=PROD_PUBLISHING_BUCKET,
+            target_bucket=target_bucket,
+            source_path=key.path + "/",
+            target_path=key.path + "/",
+        )
+
+        resolve_latest_publish(target_bucket, product)
+
+
+def clone_data_product_by_key(
+    target_bucket: str,
+    key: ProductKey,
+    include_recipe_datasets: bool = False,
+):
+    """
+    Clone specific product key and optionally the input datasets that went into it
+
+    This is a good target if working on an enhancement that involves rebuilding a specific product instance
+    """
+    copy_folder_to_dev_bucket(
+        source_bucket=PROD_PUBLISHING_BUCKET,
+        target_bucket=target_bucket,
+        source_path=key.path + "/",
+        target_path=key.path + "/",
+    )
+    if include_recipe_datasets:
+        for _index, row in publishing.get_source_data_versions(key).iterrows():
+            clone_recipe(target_bucket, row["datalibrary_name"], row["version"])
+
+
+def setup(id: str, clean: bool = False) -> str:
+    """
+    Sets up a dev bucket
+    Creates bucket if it doesn't exist.
+    If clean flag provided, deletes all objects within bucket
+    """
+    from dcpy.utils import s3
+
+    bucket = f"{DEV_BUCKET_PREFIX}{id}"
+    if bucket not in [
+        bucket["Name"] for bucket in s3.client().list_buckets()["Buckets"]
+    ]:
+        s3.client().create_bucket(Bucket=bucket, ACL=ACL)
+    else:
+        if not clean:
+            raise Exception(
+                f"bucket {bucket} already exists. specify '-c' to clean contents of bucket"
+            )
+        aws_s3_endpoint = os.environ.get("AWS_S3_ENDPOINT")
```

Comment:
one alternative possibility, curious your thoughts: What if in our s3 utils, we add delete functionality that throws an exception if pointed at the prod bucket. Would allow us to unit test it.